plugins/forms/dialogs/button.js

```diff
@@ -50,7 +50,7 @@ CKEDITOR.dialog.add( 'button', function( editor ) {
 
 			var writer = new CKEDITOR.htmlParser.basicWriter();
 			fake.writeHtml( writer );
-			var newElement = CKEDITOR.dom.element.createFromHtml( writer.getHtml(), editor.document );
+			var newElement = CKEDITOR.dom.element.createFromHtml( CKEDITOR.tools.htmlSafeByReview(writer.getHtml(), 'The writer variable is created using a safe internal function. Any attributes coming from the fake element are also controlled and internal'), editor.document );
```

Comment:
Let's figure out what `writer` is supposed to be and legacy convert if we can't convince ourselves it's safe or if it can't be sanitized?

Reply:
`writer` is defined as a (`basicWriter`)[https://sourcegraph.com/github.com/ckeditor/ckeditor4/-/blob/core/htmlparser/basicwriter.js] which doesn't seem to have any sanitization. The element being written to this writer is the selected element which must be of type button/reset/submit. I'm not sure if I entirely trust this dataflow so might need to legacy convert.

Reply:
Legacy converted